src/Formik.tsx

```diff
@@ -445,16 +445,17 @@ export class Formik<Values = FormikValues> extends React.Component<
       this.setState({ isValidating: false });
       const isValid = Object.keys(combinedErrors).length === 0;
       if (isValid) {
-        this.executeSubmit();
+        return Promise.resolve(this.executeSubmit());
       } else if (this.didMount) {
         // ^^^ Make sure Formik is still mounted before calling setState
         this.setState({ isSubmitting: false });
       }
+      return;
```

Comment:
Was this intentional to return `void` when there are validation errors? This would imply that consumer is checking the result of `submitForm()` before chaining to it

Reply:
Agree we should rather harmonize on a single return type like Promise<Result | null>

Reply:
`Promise<Result | undefined>`